decoding.py

```diff
@@ -258,8 +274,32 @@ def cross_validation_across_time(data_x, data_y, clf, add_null_data=False,
         all_probas[idxs_test] = probas
 
         preds = np.argmax(probas, -1)
-
-        accuracy = (preds == test_y[:, None]).mean(axis=0)
+        
+        if metric == "accuracy": 
+            accuracy = (preds == test_y[:, None]).mean(axis=0)
+        else:
+            # resolve metric function 
+            if isinstance(metric, str):
+                if not hasattr(sk_metrics, metric):
+                    raise ValueError(f"sklearn.metrics has no function named '{metric}'")
+                func = getattr(sk_metrics, metric)
+            elif callable(metric):
+                func = metric
+            else:
+                raise TypeError("metric must be 'accuracy', a sklearn.metrics name (str), or a callable.")
+    
+            sig = inspect.signature(func)
+            # add any extra parameters that are not preds and data_y
+            if metric_kwargs:
+                kwargs = {}
+                for k, v in metric_kwargs.items():
+                    if k in sig.parameters:
+                        kwargs[k] = v
+                        
+            # need to loop over timepoints 
+            accuracy = np.zeros(time_max)
+            for t in list(range(0, time_max)): 
+                accuracy[t] = func(data_y, all_probas[:,t], **kwargs)
```

Comment:
```suggestion
                accuracy[t] = func(data_y, all_probas[:,t], **metric_kwargs)
```
after resolving the above comments